sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2.scala

```diff
@@ -82,7 +82,9 @@ object HiveThriftServer2 extends Logging {
 
     ShutdownHookManager.addShutdownHook { () =>
       SparkSQLEnv.stop()
-      uiTab.foreach(_.detach())
+      if (uiTab != null) {
+        uiTab.foreach(_.detach())
```

Comment:
I think it's easier to just init the Option to None instead of `_` which is null.